src/silo/query_engine/exec_node/table_scan.cpp

```diff
@@ -8,68 +15,37 @@ namespace {
 
 template <typename SymbolType>
 arrow::Status appendSequences(
-   const storage::column::SequenceColumnPartition<SymbolType>& sequence_store,
+   const storage::column::SequenceColumnPartition<SymbolType>& sequence_column_partition,
    const roaring::Roaring& row_ids,
    arrow::BinaryBuilder& output_array
 ) {
+   size_t cardinality = row_ids.cardinality();
+
    std::string general_reference;
    std::ranges::transform(
-      sequence_store.metadata->reference_sequence,
+      sequence_column_partition.metadata->reference_sequence,
       std::back_inserter(general_reference),
       SymbolType::symbolToChar
    );
```

Comment:
Why don't we use this instead here?

```suggestion
   std::string general_reference = sequence_column_partition.reference_sequence_string;
```